### PR TITLE
Docs: Follow the style guide

### PR DESCRIPTION
**Pull request by hugovk**

### CODE REVIEW

File: docs/blog/2017-09-18-gatsby-modern-static-generation/index.md [hunk 1]
```diff
@@ -165,11 +165,11 @@ ops required by such a stack. (There are paid hosting options, of course, but
 that comes with a different set of problems). So I switched to Gatsby.
 
 Another option, if you want to keep Wordpress' Admin UI is to maintain your
-content, is to separate the backend from the frontend.
+content, is to separate the backend from the front end.
```
Comment (amberleyromo) [inline docs/blog/2017-09-18-gatsby-modern-static-generation/index.md]: In some of the other changes, you made it `front-end` (which I also prefer), can we use that please?

File: docs/blog/2017-09-26-embracing-graphql/index.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 title: "Embracing GraphQL — How I migrated React FAQ to Gatsby v1"
 date: "2017-09-26"
 image: "toa-heftiba-274947.jpg"
-author: "Tim Arney"
+author: "Time Arney"
```
Comment (amberleyromo) [inline docs/blog/2017-09-26-embracing-graphql/index.md]: This is a name -- please don't change this.

File: docs/blog/2017-10-05-portfolio-site-gatsby-wordpress/index.md [hunk 3]
```diff
@@ -53,8 +53,8 @@ Originally, following Scott's and the official Gatsby tutorial, I had it pulling
 my content from good ol' Markdown files. Then I saw it... too good to be true,
 the mention of connecting it to WordPress... I have been writing WordPress
 themes for almost 2 years now, so naturally this got me super excited. To have
-my back-end as WordPress (including
-[ACF](https://www.advancedcustomfields.com/)) and the front-end in React… I had
+my back end as WordPress (including
```
Comment (amberleyromo) [inline docs/blog/2017-10-05-portfolio-site-gatsby-wordpress/index.md]: Leave this, I think. "front-end", "back-end", etc.

File: docs/blog/2017-10-05-portfolio-site-gatsby-wordpress/index.md [hunk 4]
```diff
@@ -53,8 +53,8 @@ Originally, following Scott's and the official Gatsby tutorial, I had it pulling
...
+[ACF](https://www.advancedcustomfields.com/)) and the front end in React… I had
```
Comment (amberleyromo) [inline docs/blog/2017-10-05-portfolio-site-gatsby-wordpress/index.md]: `front-end`

File: docs/blog/2017-10-16-making-website-building-fun/index.md [hunk 5]
```diff
@@ -223,7 +223,7 @@ around, but it wouldn't be anything special.
 ## Reducing the complexity around the _entire_ process of building a website
 
 If we think about the process of building for the web, there are many steps
-before we get to actually coding the frontend.
+before we get to actually coding the front end.
```
Comment (amberleyromo) [inline docs/blog/2017-10-16-making-website-building-fun/index.md]: `front-end`

File: docs/blog/2017-10-20-from-wordpress-to-developing-in-react-starting-to-see-it/index.md [hunk 6]
```diff
@@ -158,7 +158,7 @@ now live at
 
 Together with my colleagues at
 [Indigo Tree](https://indigotree.co.uk "Indigo Tree: We Build Awesome Websites")
-we’re now looking at using WordPress as a backend, where clients can edit their
+we’re now looking at using WordPress as a back end, where clients can edit their
```
Comment (amberleyromo) [inline docs/blog/2017-10-20-from-wordpress-to-developing-in-react-starting-to-see-it/index.md]: `back-end`

File: docs/blog/2017-11-09-why-i-created-my-blog-with-gatsby-and-contentful/index.md [hunk 7]
```diff
@@ -255,7 +255,7 @@ form we wish.
 
 That also means that our clients can now use a proper CMS to manage all the
 content of their sites, and whenever they publish, update or delete an entry,
-the whole static frontend will be auto-generated from scratch, resulting in a
+the whole static front end will be auto-generated from scratch, resulting in a
```
Comment (amberleyromo) [inline docs/blog/2017-11-09-why-i-created-my-blog-with-gatsby-and-contentful/index.md]: `front-end`

File: docs/blog/2017-11-09-why-i-created-my-blog-with-gatsby-and-contentful/index.md [hunk 8]
```diff
@@ -271,11 +271,11 @@ its entries via an HTTP endpoint, as well as setting up webhooks that can be
 triggered when those entries are created, updated or deleted. Fortunately, most
 popular CMSs nowadays already support those features, either natively or by
 installing plugins (i.e.
-[Wordpress REST API](https://wordpress.org/plugins/rest-api/)).
+[WordPress REST API](https://wordpress.org/plugins/rest-api/)).
 
 Personally, however, I found the idea of maintaining a CMS server and a database
 just for my simple blog not very appealing. Yes, our decoupled architecture
-means that if our CMS server or database ever goes down our static frontend
+means that if our CMS server or database ever goes down our static front end
```
Comment (amberleyromo) [inline docs/blog/2017-11-09-why-i-created-my-blog-with-gatsby-and-contentful/index.md]: `front-end`

File: docs/blog/2017-12-06-gatsby-plus-contentful-plus-netlify/index.md [hunk 9]
```diff
@@ -95,19 +95,19 @@ repo.
 # Solution: Contentful + Gatsby
 
 Contentful is a hosted headless CMS with a fantastic user experience. It’s
-similar to having a backend like Wordpress, but you are fully responsible for
+similar to having a back end like Wordpress, but you are fully responsible for
```
Comment (amberleyromo) [inline docs/blog/2017-12-06-gatsby-plus-contentful-plus-netlify/index.md]: `back-end`

File: docs/blog/2017-12-06-gatsby-plus-contentful-plus-netlify/index.md [hunk 10]
```diff
@@ -95,19 +95,19 @@ repo.
...
 the front-end layer. The beauty of Contentful is threefold.
 
 - Intuitive and attractive UI
 - Simple content modeling
 - [Free tier](https://www.contentful.com/pricing/)
 
-Dealing with the back-end of Contentful is refreshing and the content modeling
+Dealing with the back end of Contentful is refreshing and the content modeling
```
Comment (amberleyromo) [inline docs/blog/2017-12-06-gatsby-plus-contentful-plus-netlify/index.md]: `back-end`

File: docs/blog/2017-12-06-gatsby-plus-contentful-plus-netlify/index.md [hunk 11]
```diff
@@ -95,19 +95,19 @@ repo.
...
 really leads the pack when compared to other headless content management
 systems. It doesn’t feel like something that just gets the job done, it’s
 actually really nice to use. They also just pushed some
 [great new changes](https://www.contentful.com/blog/2017/11/28/work-smarter-with-our-new-search-features/)
-that made it even easier to search and filter our articles on the back-end.
+that made it even easier to search and filter our articles on the back end.
```
Comment (amberleyromo) [inline docs/blog/2017-12-06-gatsby-plus-contentful-plus-netlify/index.md]: `back-end`

File: docs/blog/2017-12-07-taking-gatsby-for-a-spin/index.md [hunk 12]
```diff
@@ -49,7 +49,7 @@ Gatsby definitely delivers on speed. The whole PRPL pattern thing seems to work
 
 ### GraphQL is amazing
 
-Data from anywhere with static output. That's sort of the holy grail isn't it? Right now the data I use in this site comes from markdown files, but I can already see that switching to another content source is going to be a breeze with GraphQL. The queries are clean and readable and the Graph*i*QL IDE is the perfect helper. In a way, GraphQL allows you to separate your front-end from the type of data source.
+Data from anywhere with static output. That's sort of the holy grail isn't it? Right now the data I use in this site comes from markdown files, but I can already see that switching to another content source is going to be a breeze with GraphQL. The queries are clean and readable and the Graph*i*QL IDE is the perfect helper. In a way, GraphQL allows you to separate your front end from the type of data source.
```
Comment (amberleyromo) [inline docs/blog/2017-12-07-taking-gatsby-for-a-spin/index.md]: Leave this as `front-end`

File: docs/blog/2018-03-29-migration-from-wordpress-to-gatsby/index.md [hunk 13]
```diff
@@ -56,7 +56,7 @@ There are many factors to consider. Maturity is important as it means a big comm
...
+Gatsby is built with a lot of interesting technologies. If you are a front-end developer, you'll probably be familiar with most of these, so adoption should be quick. If not, it gives you chance to learn a lot of interesting stuff. Gatsby is powered by React. You'll be able to nicely structure your project to components and use all the existing React components there not specifically developed for Gatsby. For styling, you can attach plain old CSS stylesheet, but Gatsby offers many more choices like Sass, Glamor, Styled Components, Stylus, Less or Styled JSX. Gatsby is built on top of Webpack. And it can generate offline-ready Progressive Web App and uses GraphQL for querying data.
```
Comment (amberleyromo) [inline docs/blog/2018-03-29-migration-from-wordpress-to-gatsby/index.md]: Strange, I can't see what changed here.

File: docs/blog/2018-08-09-swag-store/index.md [hunk 14]
```diff
@@ -50,7 +50,7 @@ We love seeing how great the community looks in purple, so please post your Gats
 
 ## How the store works
 
-All of the source code for the swag store [has been open sourced][swag-source]. The front-end is a [Gatsby v2][v2] site (naturally) using [Shopify][shopify] (and their [JS Buy SDK][js-buy-sdk]) to display products and manage carts, [Auth0][auth0] to handle authentication, and a [custom Express API][swag-api] to manage discount code requests.
+All of the source code for the swag store [has been open sourced][swag-source]. The front end is a [Gatsby v2][v2] site (naturally) using [Shopify][shopify] (and their [JS Buy SDK][js-buy-sdk]) to display products and manage carts, [Auth0][auth0] to handle authentication, and a [custom Express API][swag-api] to manage discount code requests.
```
Comment (amberleyromo) [inline docs/blog/2018-08-09-swag-store/index.md]: Leave as `front-end`

File: docs/blog/2018-1-24-creative-software-destruction-new-presentation-layer/index.md [hunk 15]
```diff
@@ -9,7 +9,7 @@ tags: ["architecture", "serverless", "cdn"]
...
 
-This process of creative destruction tends to begin within the infrastructure sector but eventually has broad impact across the whole stack. Most of the attention in recent years has been focused around transforming the backend with the adoption of containers and microservices. Mobile has brought about changes to the presentation layer, but the timing is right for a more fundamental change in the presentation layer architecture — and the seeds of that change have begun.
+This process of creative destruction tends to begin within the infrastructure sector but eventually has broad impact across the whole stack. Most of the attention in recent years has been focused around transforming the back end with the adoption of containers and microservices. Mobile has brought about changes to the presentation layer, but the timing is right for a more fundamental change in the presentation layer architecture — and the seeds of that change have begun.
```
Comment (amberleyromo) [inline docs/blog/2018-1-24-creative-software-destruction-new-presentation-layer/index.md]: `back-end`

File: docs/blog/2018-2-27-why-i-upgraded-my-website-to-gatsbyjs-from-jekyll/index.md [hunk 16]
```diff
@@ -30,7 +30,7 @@ However, it would be a pain to have to manually convert images into different si
...
 
-I guess having used [webpack](https://webpack.js.org/) at work, I was pampered by this open source community where there are loaders and documented recipes for doing almost anything, granted that someone was willing to wade into the world of "JavaScript fatigue". Around the same time, [@yangshun](https://github.com/yangshun) introduced me to [Gatsby](https://www.gatsbyjs.org/), a React static site generator which seemed really fascinating. It also seemed a good opportunity for me to get my hands dirty with frontend development again.
+I guess having used [webpack](https://webpack.js.org/) at work, I was pampered by this open source community where there are loaders and documented recipes for doing almost anything, granted that someone was willing to wade into the world of "JavaScript fatigue". Around the same time, [@yangshun](https://github.com/yangshun) introduced me to [Gatsby](https://www.gatsbyjs.org/), a React static site generator which seemed really fascinating. It also seemed a good opportunity for me to get my hands dirty with front-end development again.
```
Comment (amberleyromo) [inline docs/blog/2018-2-27-why-i-upgraded-my-website-to-gatsbyjs-from-jekyll/index.md]: Make this `Webpack` while we're here?

File: docs/blog/2018-2-6-choosing-a-back-end/index.md [hunk 17]
```diff
@@ -1,11 +1,11 @@
 ---
-title: Picking a back-end for GatsbyJS
+title: Picking a back end for GatsbyJS
```
Comment (amberleyromo) [inline docs/blog/2018-2-6-choosing-a-back-end/index.md]: Leave this `back-end'

File: docs/blog/2018-2-6-choosing-a-back-end/index.md [hunk 18]
```diff
@@ -1,11 +1,11 @@
...
 date: "2018-02-06"
 author: "Ross Whitehouse"
 tags: ["headless-cms", "wordpress", "contentful", "netlify-cms", "jamstack"]
 ---
 
-A little while ago I had the six-monthly itch to redo my portfolio site, and I decided I was going to learn to use Gatsby. But that seems to be only half the battle. After you’ve built your front-end, how do you decide on the back-end? There’s so many out there!
+A little while ago I had the six-monthly itch to redo my portfolio site, and I decided I was going to learn to use Gatsby. But that seems to be only half the battle. After you’ve built your front end, how do you decide on the back end? There’s so many out there!
```
Comment (amberleyromo) [inline docs/blog/2018-2-6-choosing-a-back-end/index.md]: Leave as `front-end` and `back-end`

File: docs/blog/2018-2-6-choosing-a-back-end/index.md [hunk 19]
```diff
@@ -18,18 +18,18 @@ Gatsby pitches as a “Blazing-fast static site generator for React”. So if I
...
+So if you have any experience with React, or if you’re just getting started, this is a great choice. Gatsby doesn’t tell you how your code has to look. In fact it’s setup so you can use Markdown files for pages, but I didn’t fancy that, so I don’t. Simple as that! Likewise, if I want to scale this to build websites for other people, I don’t want to spend ages teaching them how to use Markdown, clone a Git repo and add it to the Git repo. Queue a Headless CMS.
 
-You build your site. You’ve got your SASS, you’ve got your Markdown files (or not), but everything’s blank! So now what? How do we populate it with content?
+You build your site. You’ve got your Sass, you’ve got your Markdown files (or not), but everything’s blank! So now what? How do we populate it with content?
 
-## The Next Step: A back-end
+## The Next Step: A back end
```
Comment (amberleyromo) [inline docs/blog/2018-2-6-choosing-a-back-end/index.md]: Leave as `back-end`

File: docs/blog/2018-2-6-choosing-a-back-end/index.md [hunk 20]
```diff
@@ -118,13 +118,13 @@ Prismic has a similar pricing structure to Contentful - but has a few more optio
...
+- It’s open source - anyone can download it, anyone can contribute to the Git repository and improve it, meaning it’s fully free and will be available - in one form or another - forever. I could see an issue with Contentful if the service ever stopped working. They have backups on AWS and create nightly backups on their premium plans, but the actual interface may become unavailable. Because Cockpit is open source they could go out of business or go offline for a night or just stop working altogether, and there’s a repo available with your CMS on.
 
 - It’s self-hosted - this ties into the last point and if it did all come crashing down, as long as your site is still active so will your CMS be. Great news for the tech-paranoid!
 
 ---
 
-## Conclusion - Which Back-End is right?
+## Conclusion - Which back end is right?
```
Comment (amberleyromo) [inline docs/blog/2018-2-6-choosing-a-back-end/index.md]: Leave the hyphen but agree with lower casing.

File: docs/docs/behind-the-scenes-terminology.md [hunk 21]
```diff
@@ -35,7 +35,7 @@ Contains a map of Page [path](#path) -> [Page object](#page-object).
 
 ### matchPath
 
-Think of this instead as `client matchPath`. It is ignored when creating pages during the build. But on the frontend, when resolving the page from the path ([find-path.js]()), it is used (via [reach router](https://github.com/reach/router/blob/master/src/lib/utils.js)) to find the matching page. Note that the [pages are sorted](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/internal-plugins/query-runner/pages-writer.js#L38) so that those with matchPaths are at the end, so that explicit paths are matched first.
+Think of this instead as `client matchPath`. It is ignored when creating pages during the build. But on the front end, when resolving the page from the path ([find-path.js]()), it is used (via [reach router](https://github.com/reach/router/blob/master/src/lib/utils.js)) to find the matching page. Note that the [pages are sorted](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/internal-plugins/query-runner/pages-writer.js#L38) so that those with matchPaths are at the end, so that explicit paths are matched first.
```
Comment (amberleyromo) [inline docs/docs/behind-the-scenes-terminology.md]: `front-end`

File: docs/docs/ecommerce-tutorial/index.md [hunk 22]
```diff
@@ -21,7 +21,7 @@ title: "Gatsby E-Commerce Tutorial"
 
 # Why use Gatsby for an e-commerce site?
 
-In this advanced tutorial, you’ll learn how to use Gatsby to build the UI for a basic e-commerce site that can accept payments, with Stripe as the backend for processing payments. Benefits of using Gatsby for e-commerce sites include the following:
+In this advanced tutorial, you’ll learn how to use Gatsby to build the UI for a basic e-commerce site that can accept payments, with Stripe as the back end for processing payments. Benefits of using Gatsby for e-commerce sites include the following:
```
Comment (amberleyromo) [inline docs/docs/ecommerce-tutorial/index.md]: `back-end`

File: docs/docs/ecommerce-tutorial/index.md [hunk 23]
```diff
@@ -42,7 +42,7 @@ Stripe is a payment processing service that allows you to securely collect and p
 
 There are alternatives to Stripe, like Square and Braintree, and their setup is very similar to Stripe.
 
-Stripe requires a server to process the information to make a charge, so you’ll need more than just static pages. Gatsby builds the pages for your front-end, but it won’t be able to handle the server logic that Stripe requires to process payments. That means you’ll need to set up a simple function that your Gatsby project can POST to in order to handle a payment.
+Stripe requires a server to process the information to make a charge, so you’ll need more than just static pages. Gatsby builds the pages for your front end, but it won’t be able to handle the server logic that Stripe requires to process payments. That means you’ll need to set up a simple function that your Gatsby project can POST to in order to handle a payment.
```
Comment (amberleyromo) [inline docs/docs/ecommerce-tutorial/index.md]: Leave as `front-end`

File: docs/docs/gatsby-project-structure.md [hunk 24]
```diff
@@ -28,7 +28,7 @@ Inside a Gatsby project, you may see some or all of the following folders and fi
 
 - **`/plugins`** This folder hosts any project-specific ("local") plugins that aren't published as an `npm` package. Check out the [plugin docs](/docs/plugins/) for more detail.
 
-- **`/src`** This directory will contain all of the code related to what you will see on the front-end of your site (what you see in the browser), like your site header, or a page template. “Src” is a convention for “source code”.
+- **`/src`** This directory will contain all of the code related to what you will see on the front end of your site (what you see in the browser), like your site header, or a page template. “Src” is a convention for “source code”.
```
Comment (amberleyromo) [inline docs/docs/gatsby-project-structure.md]: Leave alone as `front-end`

File: docs/docs/headless-cms.md [hunk 25]
```diff
@@ -5,11 +5,11 @@ overview: true
 
 A core feature of Gatsby is its ability to **load data from anywhere**. This is what makes Gatsby more powerful than many other static site generators that are limited to only loading content from Markdown files.
 
-A core benefit of this “data from anywhere” approach is that it allows teams to manage their content in nearly any back-end they prefer.
+A core benefit of this “data from anywhere” approach is that it allows teams to manage their content in nearly any back end they prefer.
```
Comment (amberleyromo) [inline docs/docs/headless-cms.md]: `back-end`

File: docs/docs/headless-cms.md [hunk 26]
```diff
@@ -5,11 +5,11 @@ overview: true
...
 
 Many content management systems (CMS) now support a “headless” mode, which is perfect for Gatsby sites. A headless CMS allows content creators to manage their content through a familiar admin interface, but allows developers to access the content via API endpoints, allowing for a fully customized front-end experience.
 
-Through use of [source plugins](/plugins/?=source), Gatsby has support for dozens of headless CMS options, allowing your content team the comfort and familiarity of its preferred admin interface, and your development team the improved developer experience and performance gains of using Gatsby, GraphQL, and React to build the front-end.
+Through use of [source plugins](/plugins/?=source), Gatsby has support for dozens of headless CMS options, allowing your content team the comfort and familiarity of its preferred admin interface, and your development team the improved developer experience and performance gains of using Gatsby, GraphQL, and React to build the front end.
```
Comment (amberleyromo) [inline docs/docs/headless-cms.md]: `front-end`

File: docs/docs/performance-tracing.md [hunk 27]
```diff
@@ -5,14 +5,14 @@ title: "Performance tracing"
...
-- [Tracing backend examples](/docs/performance-tracing/#tracing-backend-examples)
+- [Tracing back-end examples](/docs/performance-tracing/#tracing-backend-examples)
 - [Adding your own tracing](/docs/performance-tracing/#adding-your-own-tracing)
 
 ![Example Zipkin Trace](./images/zipkin-trace.png)
 
 ## Running Gatsby with tracing turned on
 
-Gatsby code is instrumented with OpenTracing, which is a general tracing API that is implementation agnostic. Therefore, you'll need to include and configure an OpenTracing compatible library in your application, as well as a backend to collect the trace data.
+Gatsby code is instrumented with OpenTracing, which is a general tracing API that is implementation agnostic. Therefore, you'll need to include and configure an OpenTracing compatible library in your application, as well as a back end to collect the trace data.
```
Comment (amberleyromo) [inline docs/docs/performance-tracing.md]: `back-end`

File: docs/docs/performance-tracing.md [hunk 28]
```diff
@@ -22,20 +22,20 @@ Add an [OpenTracing compatible library](https://github.com/opentracing) to your
 
 ### 2. Library configuration file
 
-Each OpenTracing library must be configured. For example, what is the URL of the tracing backend? How often should spans be sent to the backend? What service name should the trace be recorded under? Etc.
+Each OpenTracing library must be configured. For example, what is the URL of the tracing back end? How often should spans be sent to the back end? What service name should the trace be recorded under? Etc.
```
Comment (amberleyromo) [inline docs/docs/performance-tracing.md]: `back-end`

File: docs/docs/performance-tracing.md [hunk 29]
```diff
@@ -22,20 +22,20 @@ Add an [OpenTracing compatible library](https://github.com/opentracing) to your
...
 
-The configuration file is a javascript file that exports two functions. `create` and `stop`
+The configuration file is a JavaScript file that exports two functions. `create` and `stop`
 
 - **create**: Create and return an [OpenTracing compatible Tracer](https://github.com/opentracing/opentracing-javascript/blob/master/src/tracer.ts). It is called at the start of the build
-- **stop**: Called at the end of the build. Any cleanup required by the tracer should be performed here. Such as clearing out any span queues and sending them to the tracing backend.
+- **stop**: Called at the end of the build. Any cleanup required by the tracer should be performed here. Such as clearing out any span queues and sending them to the tracing back end.
```
Comment (amberleyromo) [inline docs/docs/performance-tracing.md]: `back-end`

File: docs/docs/performance-tracing.md [hunk 30]
```diff
@@ -22,20 +22,20 @@ Add an [OpenTracing compatible library](https://github.com/opentracing) to your
...
 
 ### 3. Start Gatsby with tracing turned on
 
 The above configuration file can be passed to Gatsby with the `--open-tracing-config-file` command-line option. When Gatsby is started with this option, it will load the supplied tracing configuration file, and call its `create` function. The returned Tracer will be used for tracing the build. Once the build has stopped, the configuration file's `stop` method will be called, allowing the tracing implementation to perform any cleanup.
 
-## Tracing backend examples
+## Tracing back end examples
```
Comment (amberleyromo) [inline docs/docs/performance-tracing.md]: `back-end`

File: docs/docs/performance-tracing.md [hunk 31]
```diff
@@ -22,20 +22,20 @@ Add an [OpenTracing compatible library](https://github.com/opentracing) to your
...
 
-There are many open tracing compatible backends available. Below is an example of how to hook zipkin into Gatsby
+There are many open tracing compatible back ends available. Below is an example of how to hook zipkin into Gatsby
```
Comment (amberleyromo) [inline docs/docs/performance-tracing.md]: `back-ends`

File: docs/docs/performance-tracing.md [hunk 32]
```diff
@@ -76,6 +76,6 @@ exports.sourceNodes = async ({ actions, tracing }) => {
 }
 ```
 
-With this span, you can perform any OpenTracing span operations such as [span.setTag](https://github.com/opentracing/opentracing-javascript/blob/master/src/span.ts#L89). Just make sure that the tracing backend supports these operations. You can provide an optional second span options argument to `startSpan` which will be passed to the underlying OpenTracing call.
+With this span, you can perform any OpenTracing span operations such as [span.setTag](https://github.com/opentracing/opentracing-javascript/blob/master/src/span.ts#L89). Just make sure that the tracing back end supports these operations. You can provide an optional second span options argument to `startSpan` which will be passed to the underlying OpenTracing call.
```
Comment (amberleyromo) [inline docs/docs/performance-tracing.md]: `back-end`

File: docs/blog/2018-03-29-migration-from-wordpress-to-gatsby/index.md (hunk 13 above)
Comment (hugovk) [reply docs/blog/2018-03-29-migration-from-wordpress-to-gatsby/index.md]: Hmm, GitHub doesn't do a good job of showing it!

It changes "SASS" and "LESS" to "Sass" and "Less".

File: docs/blog/2018-2-27-why-i-upgraded-my-website-to-gatsbyjs-from-jekyll/index.md (hunk 16 above)
Comment (hugovk) [reply docs/blog/2018-2-27-why-i-upgraded-my-website-to-gatsbyjs-from-jekyll/index.md]: Done!

File: docs/blog/2017-09-26-embracing-graphql/index.md (hunk 2 above)
Comment (hugovk) [reply docs/blog/2017-09-26-embracing-graphql/index.md]: Sorry, fixed!

**Comment by hugovk** [comment]
@amberleyromo Thank you for the review!

I'm following the [Gatsby Style Guide](https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/gatsby-style-guide.md#capitalize-proper-nouns), which says:

> Front-end development (adjective form with a dash) is when you’re working on the front end (noun form with no dash). The same goes with the back end, full stack, and many other compound terms.

I would personally just go for frontend and backend everywhere (like [Magento](https://magento.github.io/glossary/index.html?audience=developer#B), and ["why wait?"](https://www.theguardian.com/media/mind-your-language/2011/apr/04/language)), but is the style guide outdated and shall I update that too?

**Comment by amberleyromo** [comment]
Oof -- that's my mistake for sure. Thanks for pointing it out.

I am probably wrong. I'll defer to @shannonbux for this style question!

**Comment by hugovk** [comment]
Merge conflicts resolved.

**Comment by amberleyromo** [comment]
Hi @hugovk -- Haven't forgot, just wanted to let you know that @shannonbux will be back online in a couple days. Since this is a style question and not time sensitive, I feel it's worth leaving for her to resolve.

**Comment by hugovk** [comment]
Thanks for the update!

I've split out the non-style things into https://github.com/gatsbyjs/gatsby/pull/9155 and https://github.com/gatsbyjs/gatsby/pull/9156, then this can be rebased into something a bit smaller and more reviewable with just style changes in :)

**Comment by hugovk** [comment]
There we go, just the style changes.

**Comment by LekoArts** [comment]
@amberleyromo @shannonbux Review needed 👍

**Comment by shannonbux** [comment]
@hugovk it looks like most of @amberleyromo's requested changes were not addressed. This can be merged once all her comments are resolved!

**Comment by hugovk** [comment]
Hi @shannonbux! 

I think  @amberleyromo's requested changes are addressed by https://github.com/gatsbyjs/gatsby/pull/8873#issuecomment-427911834 (repeated below) and we'd like your feedback on what is preferred. Thanks!

---

I'm following the [Gatsby Style Guide](https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/gatsby-style-guide.md#capitalize-proper-nouns), which says:

> Front-end development (adjective form with a dash) is when you’re working on the front end (noun form with no dash). The same goes with the back end, full stack, and many other compound terms.

I would personally just go for frontend and backend everywhere (like [Magento](https://magento.github.io/glossary/index.html?audience=developer#B), and ["why wait?"](https://www.theguardian.com/media/mind-your-language/2011/apr/04/language)), but is the style guide outdated and shall I update that too?

**Comment by shannonbux** [comment]
Thanks for your patience @hugovk! The style guide might need to be updated. I was following another style guide I found with that particular advice and I'm open to being convinced otherwise! Is "frontend" and "backend" the most common way? Because I'd vote to go with most common over most "correct" :)

Looks like there is ambiguity, with most people agreeing that eventually, "front-end" and "front end" will collapse into frontend, though it's still in flux. It's hard to pin down language with new tech since the way people use it is changing:
https://english.stackexchange.com/questions/34447/is-it-front-end-frontend-or-front-end
https://www.designernews.co/stories/42206-frontend-frontend-or-front-end-developer

Wikipedia uses the rule we have in the Gatsby style guide:
https://en.wikipedia.org/wiki/Front-end_web_development

So the question is whether we ought to go with the probably outcome, "frontend," which could be the right guess or could jump the gun, OR we could stick with the current "correct" answer and then monitor whether than needs to be changed down the road. I'd vote for the latter, though I could be convinced otherwise.

**Comment by shannonbux** [comment]
So I'll merge this by the end of this Friday unless anyone has objections!

**Comment by hugovk** [comment]
Rebased to resolve merge conflict.

---

I think at this stage it's fine to pick either approach and document it. And Gatsby already has a style guide 👍

One big thing in favour of "frontend" everywhere: it's much easier all around. No need to think if it's being used as a noun ("front end") or as an adjective ("front-end development"). It's a bit of a tricky rule for people, and we see all three spellings used for both.

But either is fine by me!

**Comment by KyleAMathews** [comment]
👍 to "frontend" as we're a future-focused project ;-)

**Comment by hugovk** [comment]
How shall we update the style guide?

Currently:

> Front-end development (adjective form with a dash) is when you’re working on the front end (noun form with no dash). The same goes with the back end, full stack, and many other compound terms.

Do we want to retain the current advice for "normal" compounds, and only squash up "frontend" and "backend"?

For example:

> A full-stack developer (adjective form with a dash) works on the full stack (noun form with no dash). The same goes with many other compound terms.
>
> Use frontend for both adjective and noun forms as it's easier to maintain. For example, a frontend developer works on the  frontend. The same goes for backend.

**Comment by shannonbux** [comment]
Well, I don't think the docs actually ever mention "fullstack" in any form.
I don't think I'm in the know here enough. I trust your judgement! The
principle is to just be as consistent as possible, so maybe we use
fullstack, frontend, backend, and then just say that hyphens should be used
for other compound terms and just leave it vague since we don't have a
firmer rule yet.

**Pull request merged**